aodndata/aatams/aatams_sattag_qc_dm.py

```diff
@@ -0,0 +1,44 @@
+"""The AATAMS_SATTAG_QC_DM handler."""
+import os
+
+from aodncore.pipeline import HandlerBase, PipelineFilePublishType, FileType
+from .aatams_sattag_qc_dm_schema import AatamsSattagQcDmSchema
+
+# AATAMS DM global constants
+AATAMS_SATTAG_QC_DM_BASE = "IMOS/AATAMS/satellite_tagging/ATF_Location_QC_DM"
+
+
+def dest_path(filepath):
+    return os.path.join(AATAMS_SATTAG_QC_DM_BASE, os.path.basename(filepath))
+
+
+class AatamsSattagQcDmHandler(HandlerBase):
+    """The class for aatams sattag qc dm."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize the class by only accepting zip files."""
+        super(AatamsSattagQcDmHandler, self).__init__(*args, **kwargs)
+        self.allowed_extensions = [".zip"]
+        self.allowed_regexes = ["^.+_((?!nrt\\.)).+$"]
```

Comment:
why not `^.+_dm\\.zip$` ? This regex seems to allow anything different from a `_nrt` file

Reply:
yeap, forgot to update that too. I'm doing a refactoring since I didn't realize that we can just use a single pipeline class to handle both cases and reduce code size

Reply:
hum not too forgoet that the pipeline should be able to handle a zip file, but also a csv file that we have decided to repush for any reason

Reply:
@lbesnard - I didn't knew about that requirement - will add a csv option then. However, should I allow older files to be uploaded even though current files are newer (NRT and DM?).

Reply:
no i don't think it was, but considering how little the change is, I think it's valuable to have

> should I allow older files to be uploaded even though current files are newer

I would yes at this stage as it would require to check what is on the storage broker or in the wfs. but don't quote me. To be confirmed with @evacougnon

Reply:
>no i don't think it was, but considering how little the change is, I think it's valuable to have

Not as simple as adding to the extension list, but still doable (schema functions have to adapt to accept single file instead of a list of 7 files)

Reply:
Ahh, I found a simple solution - just short-circuiting in preprocessing - done.